oidc-service-impl/src/main/java/io/mosip/esignet/services/ConsentHelperService.java

```diff
@@ -189,6 +210,70 @@ private ConsentAction evaluateConsentAction(OIDCTransaction transaction, Consent
             log.error("Failed to hash the user consent", e);
             throw new EsignetException(ErrorConstants.INVALID_CLAIM);
         }
+        //compareing the new hash with the saved one
         return consentDetail.getHash().equals(hash) ? ConsentAction.NOCAPTURE : ConsentAction.CAPTURE;
     }
+
+    public boolean verifyConsentSignature (ConsentDetail consentDetail, OIDCTransaction transaction){
+        try {
+            String jwtToken = generateSignedObject(consentDetail);
+            if (StringUtils.isEmpty(jwtToken)) {
+                return false;
+            }
+            SignedJWT signedJWT = SignedJWT.parse(jwtToken);
+            JWSHeader header = signedJWT.getHeader();
+            String thumbPrint = header.getX509CertSHA256Thumbprint().toString();
+            String idHash = getIndividualIdHash(authorizationHelperService.getIndividualId(transaction));
+            Optional<PublicKeyRegistry> publicKeyRegistryOptional = publicKeyRegistryService.
+                    findFirstByIdHashAndThumbprintAndExpiredtimes(idHash, thumbPrint);
+            if (publicKeyRegistryOptional.isPresent()) {
+                Certificate certificate = IdentityProviderUtil.convertToCertificate(publicKeyRegistryOptional.get().getCertificate());
+                PublicKey publicKey = certificate.getPublicKey();
+                JWSVerifier verifier = new RSASSAVerifier((RSAPublicKey) publicKey);
+                log.info("signed jwt {} ", signedJWT);
+                if (signedJWT.verify(verifier)) {
+                    return true;
+                }
+            }
+            return false;
+        } catch (ParseException | JOSEException e) {
+            throw new EsignetException(ErrorConstants.INVALID_AUTH_TOKEN);
+        }
+    }
+
+    public String getIndividualIdHash (String individualId){
+        return IdentityProviderUtil.generateB64EncodedHash(ALGO_SHA3_256, individualId);
+    }
+
+    private String generateSignedObject (ConsentDetail consentDetail) throws ParseException {
+        List<String> acceptedClaims = consentDetail.getAcceptedClaims();
+        List<String> permittedScopes = consentDetail.getPermittedScopes();
+        String jws = consentDetail.getSignature();
```

Comment:
We already have SingatureValidator at the DTO validation level, why are we repeating the same step here?